projects/ssr-tests/src/ssr-testing_spec.ts

```diff
@@ -0,0 +1,85 @@
+import * as Log from './log.utils';
+import * as ProxyServer from './proxy.utils';
+import * as Ssr from './ssr.utils';
+
+const BACKEND_BASE_URL: string = process.env.CX_BASE_URL || '';
+
+describe('SSR E2E', () => {
+  let proxy: any;
+  const REQUEST_PATH = '/electronics-spa/en/USD/';
+
+  beforeAll(async () => {
+    Log.clearSsrLogFile();
+    await Ssr.startSsrServer();
+  });
+
+  afterEach(async () => {
+    await proxy.close();
+  });
+
+  afterAll(async () => {
+    await Ssr.killSsrServer();
+  });
+
+  it('should receive success response with request', async () => {
+    proxy = await ProxyServer.startProxyServer({
+      target: BACKEND_BASE_URL,
+    });
+    const response: any = await ProxyServer.sendRequest(REQUEST_PATH);
+    expect(response.statusCode).toEqual(200);
+
+    // Rendering should not complete in the first request.
+    // App should fall back to csr.
+    Log.assertMessages([
+      `Rendering started (${REQUEST_PATH})`,
+      `Request is waiting for the SSR rendering to complete (${REQUEST_PATH})`,
+    ]);
+  });
+
+  // Wait for SSR server to complete rendering.
+  it('should complete rendering within a minute', async () => {
+    await Log.waitUntilLogContainsText(`Rendering completed (${REQUEST_PATH})`);
+    Log.assertMessages([`Rendering completed (${REQUEST_PATH})`]);
+  });
```

Comment:
[MINOR]
Do I understand correctly, that we have a problem that in SSR we have a short `timeout` option (3s ?) after which a fallback to CSR happens, but the rendering of the homepage in practice takes more than 3s, right? 

It's too bad that this configuration issue is somehow reflected in our test specs, because it's not related to what we want to test.

Did you consider alternative approaches? E.g. increasing the `timeout` in SSR options? I hope a custom timeout value could be passed via Node env variable (not sure whether during the build time or when running ssr - `build:ssr` vs `run:ssr`):
https://github.com/SAP/spartacus/blob/2e6a1a41d6bb1886b52188d438446d55ed84f534/projects/storefrontapp/server.ts#L25-L28

Reply:
I removed this test as its not really checking anything important. It was to make sure the server was doing and completing rending so that we could check a cache response is returned in a following test. But I think we can move this logic to the next test which is scoped in this [ticket](https://jira.tools.sap/browse/CXSPA-4742).